src/test/java/com/jcabi/w3c/DefaultHtmlValidatorTest.java

```diff
@@ -105,22 +128,58 @@ public void throwsIOExceptionWhenValidationServerErrorOccurred()
             MkContainer container = null;
             try {
                 container = new MkGrizzlyContainer().next(
-                    new MkAnswer.Simple(status)
+                                new MkAnswer.Simple(status)
                 ).start();
                 new DefaultHtmlValidator(container.home())
-                    .validate("<html></html>");
+                                .validate("<html></html>");
             } catch (final IOException ex) {
                 caught.add(status);
             } finally {
                 container.stop();
             }
         }
-        MatcherAssert.assertThat(
-            caught,
-            Matchers.containsInAnyOrder(
-                responses.toArray(new Integer[responses.size()])
-            )
+        final Integer[] data = responses.toArray(new Integer[responses.size()]);
+        MatcherAssert.assertThat(caught, Matchers.containsInAnyOrder(data));
+    }
+
+    /**
+     * Build a response with valid result from W3C.
+     * @return Response from W3C.
+     */
+    private String validReturn() {
+        return StringUtils
+                .join(
+                      "<env:Envelope",
+                      " xmlns:env='http://www.w3.org/2003/05/soap-envelope'>",
+                      "<env:Body><m:markupvalidationresponse",
+                      " xmlns:m='http://www.w3.org/2005/10/markup-validator'>",
+                      "<m:validity>true</m:validity>",
+                      "<m:checkedby>W3C</m:checkedby>",
+                      "<m:doctype>text/html</m:doctype>",
+                      "<m:charset>UTF-8</m:charset>",
+                      "</m:markupvalidationresponse></env:Body></env:Envelope>"
         );
     }
 
+    /**
+     * Use a file to build the request.
+     * @return Request inside the file.
+     * @throws IOException if something goes wrong.
+     */
+    private String invalidHtmlResponse() throws IOException {
+        final InputStream file = DefaultHtmlValidator.class
+                        .getResourceAsStream("invalid-html-response.xml");
```

Comment:
@dmzaytsev it's not the request but the W3C response (mock). This file contains the real W3C response for "this is and invalid html" validation request.